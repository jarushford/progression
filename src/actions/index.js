export const getTrainingData = (data) => ({
  type: 'GET_TRAINING',
  data
})

export const getProjects = (data) => ({
  type: 'GET_PROJECTS',
  data
})

export const clearProjects = () => ({
  type: 'CLEAR_PROJECTS'
})

export const getAscents = (data) => ({
  type: 'GET_ASCENTS',
  data
})

export const clearAscents = () => ({
  type: 'CLEAR_ASCENTS'
})

export const toggleDiscipline = (discipline) => ({
  type: 'TOGGLE_DISCIPLINE',
  discipline
})

export const setUser = (user) => ({
  type: 'SET_USER',
  user
})

export const logoutUser = () => ({
  type: 'LOGOUT_USER'
})

export const toggleComplete = (key) => ({
  type: 'TOGGLE_COMPLETE',
  key
})
