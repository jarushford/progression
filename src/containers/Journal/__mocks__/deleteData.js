export const deleteDataThunk = jest.fn()