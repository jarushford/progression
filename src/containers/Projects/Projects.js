import React from 'react'
import { Link, withRouter } from 'react-router-dom'
import gradeConverter from '../../assets/gradeConverter'
import { connect }  from 'react-redux';
import { uid } from 'react-uid'
import { deleteProjectThunk } from '../../thunks/deleteProject'
import '../../main.scss'

function Projects({ projects, deleteProject, user }) {
  const projectsRender = projects.map(project => {
    const key = uid(project)
    return (
        <article className="project">
          <div>
          <Link
            to={`/projects/${project.name}`}
            key={key} className="project-link"
          >
            <h2 className="project-name">{project.name}</h2>
          </Link>
            <h5 className="project-grade">{gradeConverter[project.grade]}</h5>
          </div>
          <h5 className="project-location">{project.location}</h5>
          <i className="fas fa-times" onClick={() => deleteProject(project.id, user.id)}></i>
        </article>
    )
  })

  if (!user.name) {
    return <h1 className="no-user-msg">Log in or sign up to add projects</h1>
  }

  return (
    <section className="projects">
      <div className="projects-header">
        <h1 className="projects-title">PROJECTS</h1>
        <Link to='/projects/add'>
          <button className="add-project">Add Project</button>
        </Link>
      </div>
      {projectsRender}
    </section>
  )
}

const mapStateToProps = (state) => ({
  projects: state.projects,
  user: state.currentUser
})

const mapDispatchToProps = (dispatch) => ({
  deleteProject: (project_id, id) => dispatch(deleteProjectThunk(project_id, id))
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Projects))
