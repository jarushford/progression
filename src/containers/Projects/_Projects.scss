.projects-header {
  @include pageHeader();
}

.add-project {
  @include button();
}

.projects-title {
  @include pageTitle();
}

.projects {
  @include pageBody();
  padding-bottom: 5px;
}

.project-name {
  @include ascentDetails($accent, 1.7rem);
  margin: 10px 0 22px;
  transition: .3s ease;

  &:hover {
    color: $text;
    cursor: pointer;
  }
}

.project-location,
.project-grade {
  @include ascentDetails($text, 1.1rem);
  margin: 10px 15px 22px;
}

.project,
.project div {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.project {
  margin: 15px;
  border-bottom: 1px solid $ruleLine;
}

.project-link {
  text-decoration: none;
}

.project-form {
  @include form();
  margin-bottom: 40px;

  input,
  button,
  select {
    @include form-element();
  }

  label {
    color: $text;
    padding: 10px;
    display: flex;
    justify-content: space-around;
    align-items: center;
    font-size: 1.2rem;
  }

  h2 {
    @include form-title();
  }

  textarea {
    @include form-caption()
  }

  button {
    @include button();
    width: 200px;
    margin: 30px auto;
  }
}

input[type=number] {
  width: 157px;
}

@media screen and (max-width: 650px) {
  .project-form {
    width: 95%;

    input,
    button,
    select,
    textarea {
      width: 250px;
      margin: 15px auto;
    }

    input[type=number] {
      width: 100px;
      margin-left: 50px;
    }

    label {
      margin-left: 20%;
    }
  }
}

@media screen and (max-width: 380px) {
  button.add-project {
    width: 120px;
    font-size: .8rem;
    margin-left: 5px;
    margin-right: 10px;
  }

  .project-form label {
    margin-left: 8%;
  }
}

@media screen and (max-width: 360px) {
  .project-form {
    padding: 22px 26px;
  }
}