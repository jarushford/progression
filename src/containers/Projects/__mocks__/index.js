export const setCurrentProject = jest.fn()