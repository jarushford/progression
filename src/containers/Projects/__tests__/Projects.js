import React from 'react'
import { shallow } from 'enzyme'
import { mapStateToProps, mapDispatchToProps, Projects } from '../Projects'
import { deleteDataThunk } from '../../../thunks/deleteData'
import { fetchDataThunk } from '../../../thunks/fetchData'
import { setCurrentProject } from '../../../actions'

jest.mock('../../../thunks/deleteData')
jest.mock('../../../thunks/fetchData')
jest.mock('../../../actions')

describe('Projects', () => {
  let mockUser
  let mockProjects
  let mockFetch
  let mockDelete

  beforeEach(() => {
    mockUser = {
      name: 'Olaf',
      email: 'olaf@thatplace.com',
      password: 'olaf',
      id: 1
    }
    mockProjects = [ 
      {
        name: 'Nuthin But Sunshine',
        location: 'RMNP, CO',
        grade: 'V13',
        caption: '',
        id: 42,
        high_point: 5,
        moves_total: 7
      },
      {
        name: 'Top Notch',
        location: 'RMNP, CO',
        grade: 'V13',
        caption: '',
        id: 4,
        high_point: 7,
        moves_total: 7
      }
    ]
    mockFetch = jest.fn()
    mockDelete = jest.fn()
  })

  describe('Projects Component', () => {
    it('should match the snapshot', () => {
      const wrapper = shallow(<Projects user={mockUser} projects={mockProjects} />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot with no user', () => {
      const noUser = { name: '', email: '', password: '' }
      const wrapper = shallow(<Projects user={noUser} projects={mockProjects} />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot with no projects', () => {
      const noProjects = []
      const wrapper = shallow(<Projects user={mockUser} projects={noProjects} />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should fetch project data on click', async () => {
      const wrapper = shallow(<Projects user={mockUser} projects={mockProjects}  fetchData={mockFetch} />)

      await wrapper.find('.project-link').first().simulate('click')

      expect(mockFetch).toHaveBeenNthCalledWith(1, 1, 42, 'journal')
      expect(mockFetch).toHaveBeenNthCalledWith(2, 1, 42, 'milestone')
    })

    it('should delete a project on click', () => {
      const wrapper = shallow(<Projects user={mockUser} projects={mockProjects}  deleteProject={mockDelete} />)

      wrapper.find('.fa-times').first().simulate('click')

      expect(mockDelete).toBeCalledWith(42, 1, null, 'project')
    })
  })

  describe('mapStateToProps', () => {
    it('should return a props object with data from store', () => {
      const mockState = { currentUser: mockUser, projects: mockProjects }
      const expected = { user: mockUser, projects: mockProjects }

      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should return a props object with a method deleteProject', () => {
      const mockDispatch = jest.fn()
      const expected = deleteDataThunk()
      
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.deleteProject()

      expect(mockDispatch).toBeCalledWith(expected)
    })

    it('should return a props object with a method setCurrentProject', () => {
      const mockDispatch = jest.fn()
      const expected = setCurrentProject()
      
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.setCurrentProject()

      expect(mockDispatch).toBeCalledWith(expected)
    })

    it('should return a props object with a method fetchData', () => {
      const mockDispatch = jest.fn()
      const expected = fetchDataThunk()
      
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.fetchData()

      expect(mockDispatch).toBeCalledWith(expected)
    })
  })
})
