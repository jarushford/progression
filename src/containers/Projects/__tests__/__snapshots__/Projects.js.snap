// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects Projects Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Projects
    deleteProject={[MockFunction]}
    disciplineBoulder={true}
    fetchData={[MockFunction]}
    projects={
      Array [
        Object {
          "caption": "",
          "grade": 13,
          "high_point": 5,
          "id": 42,
          "location": "RMNP, CO",
          "moves_total": 7,
          "name": "Nuthin But Sunshine",
        },
        Object {
          "caption": "",
          "grade": 13,
          "high_point": 7,
          "id": 4,
          "location": "RMNP, CO",
          "moves_total": 7,
          "name": "Top Notch",
        },
      ]
    }
    setCurrentProject={[MockFunction]}
    user={
      Object {
        "email": "olaf@thatplace.com",
        "id": 1,
        "name": "Olaf",
        "password": "olaf",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="projects-header"
        >
          <h1
            className="projects-title"
          >
            PROJECTS
          </h1>
          <Link
            replace={false}
            to="/projects/add"
          >
            <button
              className="add-project"
            >
              Add Project
            </button>
          </Link>
        </div>,
        Array [
          <article
            className="project"
          >
            <div>
              <Link
                className="project-link"
                onClick={[Function]}
                replace={false}
                to="/projects/42"
              >
                <h2
                  className="project-name"
                >
                  Nuthin But Sunshine
                </h2>
              </Link>
              <h5
                className="project-grade"
              >
                V13 / 8B
              </h5>
            </div>
            <h5
              className="project-location"
            >
              RMNP, CO
            </h5>
            <i
              className="fas fa-times"
              onClick={[Function]}
            />
          </article>,
          <article
            className="project"
          >
            <div>
              <Link
                className="project-link"
                onClick={[Function]}
                replace={false}
                to="/projects/4"
              >
                <h2
                  className="project-name"
                >
                  Top Notch
                </h2>
              </Link>
              <h5
                className="project-grade"
              >
                V13 / 8B
              </h5>
            </div>
            <h5
              className="project-location"
            >
              RMNP, CO
            </h5>
            <i
              className="fas fa-times"
              onClick={[Function]}
            />
          </article>,
        ],
      ],
      "className": "projects",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="projects-title"
            >
              PROJECTS
            </h1>,
            <Link
              replace={false}
              to="/projects/add"
            >
              <button
                className="add-project"
              >
                Add Project
              </button>
            </Link>,
          ],
          "className": "projects-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "PROJECTS",
              "className": "projects-title",
            },
            "ref": null,
            "rendered": "PROJECTS",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="add-project"
              >
                Add Project
              </button>,
              "replace": false,
              "to": "/projects/add",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add Project",
                "className": "add-project",
              },
              "ref": null,
              "rendered": "Add Project",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "uid1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <Link
                className="project-link"
                onClick={[Function]}
                replace={false}
                to="/projects/42"
              >
                <h2
                  className="project-name"
                >
                  Nuthin But Sunshine
                </h2>
              </Link>
              <h5
                className="project-grade"
              >
                V13 / 8B
              </h5>
            </div>,
            <h5
              className="project-location"
            >
              RMNP, CO
            </h5>,
            <i
              className="fas fa-times"
              onClick={[Function]}
            />,
          ],
          "className": "project",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  className="project-link"
                  onClick={[Function]}
                  replace={false}
                  to="/projects/42"
                >
                  <h2
                    className="project-name"
                  >
                    Nuthin But Sunshine
                  </h2>
                </Link>,
                <h5
                  className="project-grade"
                >
                  V13 / 8B
                </h5>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <h2
                    className="project-name"
                  >
                    Nuthin But Sunshine
                  </h2>,
                  "className": "project-link",
                  "onClick": [Function],
                  "replace": false,
                  "to": "/projects/42",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Nuthin But Sunshine",
                    "className": "project-name",
                  },
                  "ref": null,
                  "rendered": "Nuthin But Sunshine",
                  "type": "h2",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "V13 / 8B",
                  "className": "project-grade",
                },
                "ref": null,
                "rendered": "V13 / 8B",
                "type": "h5",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "RMNP, CO",
              "className": "project-location",
            },
            "ref": null,
            "rendered": "RMNP, CO",
            "type": "h5",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-times",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
        ],
        "type": "article",
      },
      Object {
        "instance": null,
        "key": "uid2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <Link
                className="project-link"
                onClick={[Function]}
                replace={false}
                to="/projects/4"
              >
                <h2
                  className="project-name"
                >
                  Top Notch
                </h2>
              </Link>
              <h5
                className="project-grade"
              >
                V13 / 8B
              </h5>
            </div>,
            <h5
              className="project-location"
            >
              RMNP, CO
            </h5>,
            <i
              className="fas fa-times"
              onClick={[Function]}
            />,
          ],
          "className": "project",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  className="project-link"
                  onClick={[Function]}
                  replace={false}
                  to="/projects/4"
                >
                  <h2
                    className="project-name"
                  >
                    Top Notch
                  </h2>
                </Link>,
                <h5
                  className="project-grade"
                >
                  V13 / 8B
                </h5>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <h2
                    className="project-name"
                  >
                    Top Notch
                  </h2>,
                  "className": "project-link",
                  "onClick": [Function],
                  "replace": false,
                  "to": "/projects/4",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Top Notch",
                    "className": "project-name",
                  },
                  "ref": null,
                  "rendered": "Top Notch",
                  "type": "h2",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "V13 / 8B",
                  "className": "project-grade",
                },
                "ref": null,
                "rendered": "V13 / 8B",
                "type": "h5",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "RMNP, CO",
              "className": "project-location",
            },
            "ref": null,
            "rendered": "RMNP, CO",
            "type": "h5",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-times",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
        ],
        "type": "article",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="projects-header"
          >
            <h1
              className="projects-title"
            >
              PROJECTS
            </h1>
            <Link
              replace={false}
              to="/projects/add"
            >
              <button
                className="add-project"
              >
                Add Project
              </button>
            </Link>
          </div>,
          Array [
            <article
              className="project"
            >
              <div>
                <Link
                  className="project-link"
                  onClick={[Function]}
                  replace={false}
                  to="/projects/42"
                >
                  <h2
                    className="project-name"
                  >
                    Nuthin But Sunshine
                  </h2>
                </Link>
                <h5
                  className="project-grade"
                >
                  V13 / 8B
                </h5>
              </div>
              <h5
                className="project-location"
              >
                RMNP, CO
              </h5>
              <i
                className="fas fa-times"
                onClick={[Function]}
              />
            </article>,
            <article
              className="project"
            >
              <div>
                <Link
                  className="project-link"
                  onClick={[Function]}
                  replace={false}
                  to="/projects/4"
                >
                  <h2
                    className="project-name"
                  >
                    Top Notch
                  </h2>
                </Link>
                <h5
                  className="project-grade"
                >
                  V13 / 8B
                </h5>
              </div>
              <h5
                className="project-location"
              >
                RMNP, CO
              </h5>
              <i
                className="fas fa-times"
                onClick={[Function]}
              />
            </article>,
          ],
        ],
        "className": "projects",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="projects-title"
              >
                PROJECTS
              </h1>,
              <Link
                replace={false}
                to="/projects/add"
              >
                <button
                  className="add-project"
                >
                  Add Project
                </button>
              </Link>,
            ],
            "className": "projects-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "PROJECTS",
                "className": "projects-title",
              },
              "ref": null,
              "rendered": "PROJECTS",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="add-project"
                >
                  Add Project
                </button>,
                "replace": false,
                "to": "/projects/add",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Add Project",
                  "className": "add-project",
                },
                "ref": null,
                "rendered": "Add Project",
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "uid1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <Link
                  className="project-link"
                  onClick={[Function]}
                  replace={false}
                  to="/projects/42"
                >
                  <h2
                    className="project-name"
                  >
                    Nuthin But Sunshine
                  </h2>
                </Link>
                <h5
                  className="project-grade"
                >
                  V13 / 8B
                </h5>
              </div>,
              <h5
                className="project-location"
              >
                RMNP, CO
              </h5>,
              <i
                className="fas fa-times"
                onClick={[Function]}
              />,
            ],
            "className": "project",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    className="project-link"
                    onClick={[Function]}
                    replace={false}
                    to="/projects/42"
                  >
                    <h2
                      className="project-name"
                    >
                      Nuthin But Sunshine
                    </h2>
                  </Link>,
                  <h5
                    className="project-grade"
                  >
                    V13 / 8B
                  </h5>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <h2
                      className="project-name"
                    >
                      Nuthin But Sunshine
                    </h2>,
                    "className": "project-link",
                    "onClick": [Function],
                    "replace": false,
                    "to": "/projects/42",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Nuthin But Sunshine",
                      "className": "project-name",
                    },
                    "ref": null,
                    "rendered": "Nuthin But Sunshine",
                    "type": "h2",
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "V13 / 8B",
                    "className": "project-grade",
                  },
                  "ref": null,
                  "rendered": "V13 / 8B",
                  "type": "h5",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "RMNP, CO",
                "className": "project-location",
              },
              "ref": null,
              "rendered": "RMNP, CO",
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-times",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
          ],
          "type": "article",
        },
        Object {
          "instance": null,
          "key": "uid2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <Link
                  className="project-link"
                  onClick={[Function]}
                  replace={false}
                  to="/projects/4"
                >
                  <h2
                    className="project-name"
                  >
                    Top Notch
                  </h2>
                </Link>
                <h5
                  className="project-grade"
                >
                  V13 / 8B
                </h5>
              </div>,
              <h5
                className="project-location"
              >
                RMNP, CO
              </h5>,
              <i
                className="fas fa-times"
                onClick={[Function]}
              />,
            ],
            "className": "project",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    className="project-link"
                    onClick={[Function]}
                    replace={false}
                    to="/projects/4"
                  >
                    <h2
                      className="project-name"
                    >
                      Top Notch
                    </h2>
                  </Link>,
                  <h5
                    className="project-grade"
                  >
                    V13 / 8B
                  </h5>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <h2
                      className="project-name"
                    >
                      Top Notch
                    </h2>,
                    "className": "project-link",
                    "onClick": [Function],
                    "replace": false,
                    "to": "/projects/4",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Top Notch",
                      "className": "project-name",
                    },
                    "ref": null,
                    "rendered": "Top Notch",
                    "type": "h2",
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "V13 / 8B",
                    "className": "project-grade",
                  },
                  "ref": null,
                  "rendered": "V13 / 8B",
                  "type": "h5",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "RMNP, CO",
                "className": "project-location",
              },
              "ref": null,
              "rendered": "RMNP, CO",
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-times",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
          ],
          "type": "article",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Projects Projects Component should match the snapshot with no projects 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Projects
    deleteProject={[MockFunction]}
    disciplineBoulder={true}
    fetchData={[MockFunction]}
    projects={Array []}
    setCurrentProject={[MockFunction]}
    user={
      Object {
        "email": "olaf@thatplace.com",
        "id": 1,
        "name": "Olaf",
        "password": "olaf",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="projects-header"
        >
          <h1
            className="projects-title"
          >
            PROJECTS
          </h1>
          <Link
            replace={false}
            to="/projects/add"
          >
            <button
              className="add-project"
            >
              Add Project
            </button>
          </Link>
        </div>,
        <h1
          className="no-data-msg"
        >
          You have no saved projects
        </h1>,
      ],
      "className": "projects",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="projects-title"
            >
              PROJECTS
            </h1>,
            <Link
              replace={false}
              to="/projects/add"
            >
              <button
                className="add-project"
              >
                Add Project
              </button>
            </Link>,
          ],
          "className": "projects-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "PROJECTS",
              "className": "projects-title",
            },
            "ref": null,
            "rendered": "PROJECTS",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="add-project"
              >
                Add Project
              </button>,
              "replace": false,
              "to": "/projects/add",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add Project",
                "className": "add-project",
              },
              "ref": null,
              "rendered": "Add Project",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "You have no saved projects",
          "className": "no-data-msg",
        },
        "ref": null,
        "rendered": "You have no saved projects",
        "type": "h1",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="projects-header"
          >
            <h1
              className="projects-title"
            >
              PROJECTS
            </h1>
            <Link
              replace={false}
              to="/projects/add"
            >
              <button
                className="add-project"
              >
                Add Project
              </button>
            </Link>
          </div>,
          <h1
            className="no-data-msg"
          >
            You have no saved projects
          </h1>,
        ],
        "className": "projects",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="projects-title"
              >
                PROJECTS
              </h1>,
              <Link
                replace={false}
                to="/projects/add"
              >
                <button
                  className="add-project"
                >
                  Add Project
                </button>
              </Link>,
            ],
            "className": "projects-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "PROJECTS",
                "className": "projects-title",
              },
              "ref": null,
              "rendered": "PROJECTS",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="add-project"
                >
                  Add Project
                </button>,
                "replace": false,
                "to": "/projects/add",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Add Project",
                  "className": "add-project",
                },
                "ref": null,
                "rendered": "Add Project",
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "You have no saved projects",
            "className": "no-data-msg",
          },
          "ref": null,
          "rendered": "You have no saved projects",
          "type": "h1",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Projects Projects Component should match the snapshot with no user 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Projects
    deleteProject={[MockFunction]}
    disciplineBoulder={true}
    fetchData={[MockFunction]}
    projects={
      Array [
        Object {
          "caption": "",
          "grade": 13,
          "high_point": 5,
          "id": 42,
          "location": "RMNP, CO",
          "moves_total": 7,
          "name": "Nuthin But Sunshine",
        },
        Object {
          "caption": "",
          "grade": 13,
          "high_point": 7,
          "id": 4,
          "location": "RMNP, CO",
          "moves_total": 7,
          "name": "Top Notch",
        },
      ]
    }
    setCurrentProject={[MockFunction]}
    user={
      Object {
        "email": "",
        "name": "",
        "password": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Log in or sign up to add projects",
      "className": "no-user-msg",
    },
    "ref": null,
    "rendered": "Log in or sign up to add projects",
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Log in or sign up to add projects",
        "className": "no-user-msg",
      },
      "ref": null,
      "rendered": "Log in or sign up to add projects",
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
